cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/ScopeManager.kt

```diff
@@ -1084,9 +977,21 @@ class ScopeManager : ScopeProvider {
         predicate: ((Declaration) -> Boolean)? = null,
     ): List<Declaration> {
         val (scope, n) = extractScope(name, location, startScope)
+
+        // We need to differentiate between a qualified and unqualified lookup. We have a qualified
+        // lookup, if the scope is not null. In this case we need to stay within the specified scope
         val list =
-            scope?.lookupSymbol(n.localName, predicate = predicate)?.toMutableList()
-                ?: mutableListOf()
+            // TODO(oxisto): extractScope does not return null in all cases, so we need to make sure
+            //  that the returned scope is NOT our startscope
```

Comment:
I have to re-check if this is actually still a problem, because I am actually now returning null if the scope is not successfully extracted, so my first guess is that this behaviour is fixed.

Reply:
This should be solved, removing this comment